go-controller/pkg/node/gateway_init.go

```diff
@@ -162,9 +162,13 @@ func getGatewayNextHops() ([]net.IP, string, error) {
 	return gatewayNextHops, gatewayIntf, nil
 }
 
-// getDPUHostPrimaryIPAddresses returns the DPU host IP/Network based on K8s Node IP
-// and DPU IP subnet overriden by config config.Gateway.RouterSubnet
-func getDPUHostPrimaryIPAddresses(k8sNodeIP net.IP, ifAddrs []*net.IPNet) ([]*net.IPNet, error) {
```

Comment:
This function is still used in `updateInterfaceIPAddresses` in `gateway.go`. With this change, we can update that function.

Reply:
The tricky part is that #3480 also alter `getDPUHostPrimaryIPAddresses` and this is how it should be, but I'm not sure which PR will get in 1st.

Reply:
Let's pick this PR to get in first.